synapse/lib/aha.py

```diff
@@ -107,6 +123,126 @@ async def signUserCsr(self, csrtext, signas=None):
         await self._reqUserAllowed(('aha', 'csr', 'user'))
         return await self.cell.signUserCsr(csrtext, signas=signas)
 
+    @s_cell.adminapi()
+    async def addAhaSvcProv(self, name, provinfo=None):
+        '''
+        Provision the given relative service name within the configured network name.
+        '''
+        return await self.cell.addAhaSvcProv(name, provinfo=provinfo)
+
+    @s_cell.adminapi()
+    async def delAhaSvcProv(self, iden):
+        '''
+        Remove a previously added provisioning entry by iden.
+        '''
+        return await self.cell.delAhaSvcProv(iden)
+
+    @s_cell.adminapi()
+    async def addAhaUserEnroll(self, name, userinfo=None, again=False):
+        '''
+        Create and return a one-time user enroll key.
+        '''
+        return await self.cell.addAhaUserEnroll(name, userinfo=userinfo, again=again)
+
+    @s_cell.adminapi()
+    async def delAhaUserEnroll(self, iden):
+        '''
+        Remove a previously added enrollment entry by iden.
+        '''
+        return await self.cell.delAhaUserEnroll(iden)
+
+class ProvDmon(s_daemon.Daemon):
+
+    async def __anit__(self, aha):
+        self.aha = aha
+        await s_daemon.Daemon.__anit__(self)
+
+    async def _getSharedItem(self, name):
```

Comment:
do you think these should have some expiration time on them, even if its really long?